File: bitframe-utils/kotlinx-collections-interoperable/src/jsTest/resources/me.js
module.exports = {
    name: "Anderson",
    greet: function () {
        return "Hello Lamax";
    },
    doStuff: function(iterable) {
        const arr = []
        for(const obj of iterable) {
            arr.push(obj)
        }
        return arr;
    },
    convertToArray: function(list) {
        return list.toArray();
    },
    addUnbounded: function(calc,number) {
        console.log(Object.assign({},calc))
        const { plus } = calc
        console.log(plus)
        return plus(number)
    }
}

File: bitframe-utils/kotlinx-collections-interoperable/src/jsTest/kotlin/Me.kt
external interface Me {
    val name: String
    val greet: () -> Unit
    fun <T> doStuff(array: Array<T>): Array<T>
    fun <T> doStuff(list: List<T>): Array<T>
    fun <T> convertToArray(list: List<T>): Array<T>
    fun addUnbounded(calc: Calc, number: Int): Int
}